src/react-imgix.jsx

```diff
@@ -95,6 +88,7 @@ function buildSrc({
   disableSrcSet,
   imgixParams,
   disableQualityByDPR,
+  sizes,
```

Comment:
What does sizes do here?

Reply:
There's this test:
https://github.com/imgix/react-imgix/blob/ae9bcfb773dcb92f53319cea01a1c02ba8cb51bc/test/unit/react-imgix.test.jsx#L669

```jsx
it("responsive width should overwrite the width query parameter correctly", () => {
  sut = shallow(
    <Imgix src={"https://mysource.imgix.net/demo.png?w=333"} sizes="100wv" />
  );

  expect(sut.props().src).toContain("w=333");
  expect(sut.props().srcSet).not.toContain("w=333"); // <---- this check will fail if we don't check if sizes was passed.
});
```

I (very) briefly touched on this in the commit message:
>Prior to this PR, there seems to have been a little error when handling height and width values in the presence of sizes.

It seems like we want different behavior when `sizes` is passed, so that get's explicitly handled now. However, tbh this feels like a bit of a hack, so I'm open to any suggestions here.

Reply:
Okay, interesting. I'm unaware of any context on this "feature" so I'm struggling a bit to understand why we would want to change this behaviour based on whether sizes was passed or not. Do we do something along the lines of checking if sizes is passed to determine whether this is a responsive image or not.

Use your best judgement here because you have the most context, but this doesn't seem like something we need to explicitly "support"

Reply:
Yeah, I thought this bit was interesting as well. I think the prior implementation shows that this is somewhat related to the destructuring below. This line is from main:

https://github.com/imgix/react-imgix/blob/0d52511b4dddfc0d314988d12600a38751cb857f/src/react-imgix.jsx#L138

I think what was going on here is that `w width height` are all removed/excluded from the `imgixParams`. So when this test runs:

https://github.com/imgix/react-imgix/blob/ae9bcfb773dcb92f53319cea01a1c02ba8cb51bc/test/unit/react-imgix.test.jsx#L669

Specially this portion:

https://github.com/imgix/react-imgix/blob/ae9bcfb773dcb92f53319cea01a1c02ba8cb51bc/test/unit/react-imgix.test.jsx#L675

The test passes under the old implementation, but fails under the new because of the way width/height params are being passed and not being excluded. This test seems to be, "when an initial width is given, but `sizes` is also given, omit the initial width and create a width-based srcset."

```jsx
<Imgix src={"https://mysource.imgix.net/demo.png?w=333"} sizes="100wv" />
```

The only way to tell from the above that the srcset URLs should not contain a `w=333`(but should contain 31 other widths) is the `sizes` prop.